appium/options/android/common/adb/adb_options.py

```diff
@@ -0,0 +1,367 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from appium.options.base_options import OptionsDescriptor
+from appium.options.transformers import DurationTransformer
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+
+class AdbExecTimeoutOption(SupportsCapabilities):
+    ADB_EXEC_TIMEOUT = 'adbExecTimeout'
+    
+    _transform_duration_get = DurationTransformer.transform_duration_get
+    _transform_duration_set = DurationTransformer.transform_duration_set
+    
+    adb_exec_timeout = OptionsDescriptor('ADB_EXEC_TIMEOUT', _transform_duration_get, _transform_duration_set)
```

Comment:
I am still not convinced about passing uppercase characters as `name` argument. It is used as an argument for set/get_capability and there we expect a camel-case cap name

Reply:
generic types must properly be provided for each descriptor. You can run `make mypy` to verify typing locally

Reply:
> I am still not convinced about passing uppercase characters as `name` argument. It is used as an argument for set/get_capability and there we expect a camel-case cap name

Yes.. Thanks for identifying this bug. I went through the `set_capability` once again and understood that we are creating a key with camelcase in `_caps` dictionary

```
    def set_capability(self: T, name: str, value: Any) -> T:
        w3c_name = name if name in self.W3C_CAPABILITY_NAMES or ':' in name else f'{APPIUM_PREFIX}{name}'
        if value is None:
            if w3c_name in self._caps:
                del self._caps[w3c_name]
        else:
            self._caps[w3c_name] = value
        return self 
```

So yes, I should be passing the value of `ADB_EXEC_TIMEOUT` below is the correct one.

```
class AdbExecTimeoutOption(SupportsCapabilities):
    ADB_EXEC_TIMEOUT = 'adbExecTimeout'
    
    _transform_duration_get = DurationTransformer.transform_duration_get
    _transform_duration_set = DurationTransformer.transform_duration_set
    
    adb_exec_timeout = OptionsDescriptor(ADB_EXEC_TIMEOUT, _transform_duration_get, _transform_duration_set)
```

In the corrected version above , I have remove the quotes from `ADB_EXEC_TIMEOUT` while calling `OptionsDescriptor`. Now the actual value of `cls` variable `ADB_EXEC_TIMEOUT` will be passed, which is `adbExecTimeout`. I will change this in all other options.